docs/content/addons/cluster-autoscaler.md

```diff
@@ -0,0 +1,40 @@
++++
+title = "Cluster Autoscaler"
++++
+
+By leveraging CAPI cluster lifecycle hooks, this handler deploys [Cluster Autoscaler](https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler/cloudprovider/clusterapi) on the new cluster via `ClusterResourceSets`
+at the `AfterControlPlaneInitialized` phase.
+
+Deployment of Cluster Autoscaler is opt-in via the  [provider-specific cluster configuration]({{< ref ".." >}}).
+
+The hook creates a `ClusterResourceSet` to deploy the NFD resources.
+
+## Example
+
+To enable deployment of Cluster Autoscaler on a cluster, specify the following values:
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  name: <NAME>
+spec:
+  topology:
+    variables:
+      - name: clusterConfig
+        value:
+          addons:
+            nfd:
+              strategy: ClusterResourceSet
+    workers:
+      machineDeployments:
+        - class: default-worker
+          metadata:
+            annotations:
+              # Set the following annotations to configure the Cluster Autoscaler
+              # The initial MachineDeployment will have 1 Machine
+              cluster.x-k8s.io/cluster-api-autoscaler-node-group-max-size: "3"
+              cluster.x-k8s.io/cluster-api-autoscaler-node-group-min-size: "1"
```

Comment:
I feel like these should be set via the CRE variables which will patch with these annotations. WDYT? That would maek usage of CA easier, not having to know these specific annotations.

Reply:
That would mean adding properties on the `workerConfig` variable I think.

Reply:
What would the variables contain? Maybe if the autoscaler addon is enabled we automatically patch the Cluster by removing the replicas and adding these annotations?

Does this make sense to be in a webhook instead?

Reply:
Variables would contain min and max, patch would remove the replicas and set the annotations.

Reply:
Did some more testing today, I may be missing something but I didn't see handlers getting called for `MachineDeployments`, do we need to configure something else here?

Reply:
😞 Looks as though my idea was wrong, we can't patch `MachineDeployments`, only template types. So I guess we will have to go back to the `metadata.annotations` on cluster `spec.workers.machineDeployments[*]` and we can't patch the replicas here either. This means we will have to leave it up to the user and provide a clean UX to enable/disable in the UI/CLI.

Reply:
Thinking about this UX and reading https://github.com/kubernetes-sigs/cluster-api/blob/6fede8c8cbc83e108aa6a5bc4507ed3aae096105/api/v1beta1/machinedeployment_webhook.go#L278-L306, I propose that we only set the autoscaler annotations on the machinedeployments rather than the replicas field. By default this can be set to the number of replicas and we can enable autoscaling later by simply enabling the addon and changing the min/max annotations. WDYT?

Reply:
Great minds! That's exactly what you were doing in [`0044549` (#423)](https://github.com/d2iq-labs/capi-runtime-extensions/pull/423/commits/00445497254449eaf2e279c624f9674a21cd3069#diff-e520ef54af4a462ca097950d967c884c9775bc52571f31e56ce18d6d87241a3fL35-L42)

Reply:
🧠 Ok so let's basically revert the last commit and have users opt-in to cluster-autoscaler and update the replicas by manipulating the annotations in the Cluster?

Reply:
Yeah, annotations on the MachineDeployment